src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs

```diff
@@ -416,6 +418,45 @@ public async Task CallActivityAsync(DispatchMiddlewareContext dispatchContext, F
                 throw new InvalidOperationException($"An activity was scheduled but no {nameof(TaskScheduledEvent)} was found!");
             }
 
+            if (!string.IsNullOrEmpty(scheduledEvent.Name) && scheduledEvent.Name.StartsWith("BuiltIn::HttpActivity"))
```

Comment:
Instead of manually invoking the activity here, can we test if it starts with `BuiltIn::`, and if so, we can just continue to the next call in the chain:

``` CSharp
if (scheduledEvent.Name?.StartsWith("BuiltIn::", StringComparison.OrdinalIgnoreCase) ?? false)
{
    await next();
    return;
}
```

Reply:
I tested it, but wasn't able to get that working.

Reply:
Do you know why it didn't work? Were you able to debug into the rest of the pipeline?

Reply:
When I manually invoked the activity, I was serializing the output from `shim.RunAsync()` before setting it as the result. If I use the code provided, the output is throwing an error when it deserializes - `"One or more errors occurred. (The JSON value could not be converted to System.String. Path: $ | LineNumber: 0 | BytePositionInLine: 1.)",` because I think it's not serialized. Do you know if there's a way to serialize the output before it is returned?

[This line](https://github.com/Azure/azure-functions-durable-extension/blob/9f750ebc77fda1f52c5042dcdb41e6eb52ee7fdb/src/WebJobs.Extensions.DurableTask/ProtobufUtils.cs#L121) is the last place a breakpoint is hit before an exception is thrown.

Reply:
I tested this change locally and did not encounter issues. Maybe it had to happen along with JSON changes at the same time? We can sync tomorrow to see what is going on.